flight/PiOS/STM32F4xx/pios_video.c

```diff
@@ -367,15 +367,34 @@ void PIOS_Video_Init(const struct pios_video_cfg *cfg)
 	TIM_SelectOnePulseMode(cfg->hsync_capture.timer, TIM_OPMode_Single);
 	TIM_SelectSlaveMode(cfg->hsync_capture.timer, TIM_SlaveMode_Trigger);
 
+	uint16_t tmpccer = cfg->hsync_capture.timer->CCER;
+
 #ifdef PIOS_VIDEO_HSYNC_FALLING_EDGE
 	/* Unfortunately not really a stdperiph function for this. */
+	if (cfg->hsync_capture.timer_chan == TIM_Channel_1) {
+		tmpccer &= (uint16_t)~(TIM_CCER_CC1NP);
+		tmpccer |= (uint16_t)(TIM_CCER_CC1P);
+	} else if (cfg->hsync_capture.timer_chan == TIM_Channel_2) {
+		tmpccer &= (uint16_t)~(TIM_CCER_CC2NP);
+		tmpccer |= (uint16_t)(TIM_CCER_CC2P);
+	}
+#endif
 
-	uint16_t tmpccer = cfg->hsync_capture.timer->CCER;
-	tmpccer &= (uint16_t)~(TIM_CCER_CC1NP);
-	tmpccer |= (uint16_t)(TIM_CCER_CC1P);
-	cfg->hsync_capture.timer->CCER = tmpccer;
+#ifdef PIOS_VIDEO_INPUT_FILTER
+	if (cfg->hsync_capture.timer_chan == TIM_Channel_1) {
```

Comment:
Would appreciate a small comment explaining what IC1F is/does so people don't need to refer to the STM32 ref. manual. :)